### ISSUE TITLE
Parser request: semantic versions

### ISSUE THREAD

**Issue by rpdelaney** [kellyjonbrazil/jc#313]
See https://semver.org for the spec. A python-compatible [regex](https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string) is provided: https://regex101.com/r/Ly7O1x/3/

Edit: There's another regex on the page with explicitly numbered capture groups: https://regex101.com/r/vkijKf/1/

**Comment by kellyjonbrazil** [comment]
Thanks for the parser suggestion! Yes, I think that is a great idea and is easy enough to make it into the next release.

**Comment by kellyjonbrazil** [comment]
I have a working version of the parser in the `dev` branch. I still need to write tests, but I think it looks pretty good.

Here is a sample output:
```
$ echo 1.2.3-rc.1+44837 | jc --semver -p
{
  "major": 1,
  "minor": 2,
  "patch": 3,
  "prerelease": "rc.1",
  "build": "44837"
}
```
You can test this by copying the parser file to your plugin folder:
https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/semver.py

- Linux/unix: `$HOME/.local/share/jc/jcparsers`
- macOS: `$HOME/Library/Application Support/jc/jcparsers`
- Windows: `$LOCALAPPDATA\jc\jc\jcparsers`

Let me know if you run into any issues!

**Comment by rpdelaney** [comment]
Neat. The tests speak for themselves. However, `--debug` doesn't seem to do anything useful with this parser (yet?)

**Comment by kellyjonbrazil** [comment]
Yeah, that's right - it's kinda hard to get a python exception for this parser since it handles non-semver strings by just returning an empty object. If we really wanted to we could force raise an exception, but this is such a simple parser I don't think that's necessary.

**Comment by kellyjonbrazil** [comment]
Released in `jc` v1.22.2.
